src/Infrastructure/Persistence/RepositoryAsync.cs

```diff
@@ -57,8 +57,17 @@ public Task<int> GetCountAsync<T>(Expression<Func<T, bool>>? expression = null,
         return query.AsNoTracking().CountAsync(cancellationToken);
     }
 
```

Comment:
Seems like we've been doing some double work here... I've also just refactored and cleaned up RepositoryAsync as part of #237.
See my fork [here ](https://github.com/fretje/dotnet-webapi-boilerplate/blob/caching/src/Infrastructure/Persistence/RepositoryAsync.cs) for the result.